test/basics.spec.js

```diff
@@ -264,12 +264,11 @@ describe('MockAdapter basics', function() {
 
   it('can pass headers to match to a handler', function() {
     var headers = {
-      Accept: 'application/json, text/plain, */*',
       'Content-Type': 'application/x-www-form-urlencoded',
       'Header-test': 'test-header'
     };
 
-    mock.onPost('/withHeaders', undefined, headers).reply(200);
+    mock.onPost('/withHeaders', undefined, { 'Header-test': 'test-header' }).reply(200);
```

Comment:
so we are passing more headers than specified in mock and that should still match(so it validate both filtering by headers and that fact check is partial not strict)